lib/AST/ASTImporter.cpp

```diff
@@ -6081,6 +6082,31 @@ Expr *ASTNodeImporter::VisitUnaryExprOrTypeTraitExpr(
                                           Importer.Import(E->getRParenLoc()));
 }
 
+Expr *ASTNodeImporter::VisitTypeTraitExpr(TypeTraitExpr *E) {
+  QualType T = Importer.Import(E->getType());
+  if (T.isNull())
+    return nullptr;
+
+  unsigned NumArgs = E->getNumArgs();
+  llvm::SmallVector<TypeSourceInfo *, 2> ToArgTypes(NumArgs);
+  for (unsigned AI = 0, AE = NumArgs; AI != AE; ++AI) {
```

Comment:
In case of iterator it is a great way to not always reevaluate end() but the loop variables are numbers here so I think
`for (unsigned AI = 0; AI < NumArgs; ++AI)`
would be more readable. I dont see the point of using a new variable AE for that.